docs/guides/chip_tool_guide.md

```diff
@@ -1393,3 +1393,244 @@ $ ./chip-tool any read-by-id <cluster-ids> <attribute-ids> <destination-id> <end
     ```
     ./chip-tool any read-by-id 0xFFFFFFFF 0xFFFFFFFF 1 0xFFFF
     ```
+
+<hr>
+
+## Saving users and credentials on door lock devices
+
+Matter door lock devices can store pools of users and credentials that allow you
+to configure different access scenarios. Each user and credential in the pool
+has an index value. Additionally, each user has a list of Occupied Credentials.
+
+By default, each door lock device comes with no user or credential defined, but
+it reserves several slots in both pools that can be populated with new users and
+credentials, up to the value specified in the the `NumberOfTotalUsersSupported`
+attribute and the `NumberOfCredentialsSupportedPerUser` attribute, respectively.
+
+All communication between users and credentials happens only using their
+respective index values. No other information is shared between both pools.
+
+The CHIP Tool lets you add users and credentials on door lock devices and
+securely match their indexes to one another. This is an optional feature
+available only when working with devices that implement the `doorlock` cluster.
+
+> **Note:** Users and credentials can only be modified by whoever has the right
+> permissions, as specified in the Access Control List.
+
+To save credentials and users, you need to complete the following steps,
+described in detail in the following sections:
+
+1. Set up a user on the device.
+1. Assign a credential for the newly created user.
+
+### Step 1: Set up a user
+
+To set up a user on a door lock device with the CHIP Tool, use the following
+command pattern:
+
+```
+$ ./chip-tool doorlock set-user <operation-type> <user-index> <user-name> <user-unique-id> <user-status> <user-type> <credential-rule> <destination-id> <endpoint-id> --timedInteractionTimeoutMs <ms_value>
+```
+
+In this command:
+
+-   _<operation-type\>_ is one of the available types of operation for the user:
+
+    -   `Add` - This operation sets a new user in the slot at _<user-index\>_.
+    -   `Clear` - This operation removes an existing user from the slot at
+        _<user-index\>_.
+    -   `Modify` - This operation modifies an existing user at the slot at
+        _<user-index\>_.
+
+-   _<user-index\>_ is the index value of the user, between `1` and the maximum
+    value you set in the `NumberOfTotalUsersSupported` attribute. Setting the
+    user index to `0` will cause an error.
+-   _<user-name\>_ is the name of the user, which can have maximum 10 bytes of
+    size. Can be set to `null`.
+-   _<user-unique-id\>_ is a 4-byte number that describes the unique user ID.
+    Can be set to `null`.
+-   _<user-status\>_ can be set to `null` or to one of the following values:
+
+    -   `0` (`Available`) - This status indicates that the given user slot is
+        available for modification on the device. This value is selected is you
+        set `null` for the variable.
```

Comment:
"if you" instead of "is you"?

More importantly, this statement is just wrong. You can set UserStatus to null to do whatever the spec says for null. For an add, null means OccupiedEnabled. For a Modify null means "do not change the status".
```suggestion
        available for modification on the device.
```